classes/PaymentModule.php

```diff
@@ -212,6 +212,20 @@ public function validateOrder(
         if (!isset($this->context)) {
             $this->context = Context::getContext();
         }
+        $products = $this->context->cart->getProducts();
+
+        foreach ($products as $product) {
+            $availableOutOfStock = Product::isAvailableWhenOutOfStock($product['out_of_stock']);
+            $productQuantity = Product::getQuantity($product['id_product'], $product['id_product_attribute']);
+
+            if ($productQuantity < 0 && !$availableOutOfStock) {
+                $psShoppingCart = Module::getInstanceByName('ps_shoppingcart');
```

Comment:
this is so so so hardcoded... i don't think that's a good idea

Reply:
Indeed, we can't accept so strong dependencies between the core and a module.

Did you want to redirect to the cart page?